dh_alter_patient_data.sql

```diff
@@ -0,0 +1,14 @@
+# alter the patient_data table
+alter table patient_data
+ADD parent_lname varchar(255),
+ADD parent_fname varchar(255),
+ADD sparent_lname varchar(255),
+ADD sparent_fname varchar(255),
+ADD auth_until_date date,
+ADD progress_due_date date,
+ADD ifsp_review_date date,
+ADD annual_ifsp_review_date date,
+ADD aging_out_date date,
+ADD trans_conf_date date,
+ADD patient_type varchar(30),
+ADD active varchar(4);
```

Comment:
Do you think any of these patient_data settings will be helpful for general users? If so, let me know, and we can further discuss best strategy to incorporate them.
(Note there is a Guardian section in layouts, albeit that only takes in 1 parent)

Reply:
the dates are not that useful outside of my wife's practice.  There may be way to use reminders to accomplish the same needs, but I didn't have those figured out yet.  My hammer found a nail...

The active flag would be useful to hide some patients by default from clinicians, while keeping them in the system for administration and billing.  She has a pediatric practice, so her patients age out of various funding sources such as CMS, etc.

Her therapy practice only provides a few different categories of treatments, and we will be using the appointment type field to characterize them.  A child that receives one kind of these therapies will continue with them, often for years.  This is a different use case than a GP or practice with a lot of diagnoses.  We needed a default per child, rather than per clinician or facility.  Thinking about it, I need to look at using facilities for these maybe...

The patient type is the pc_constant_id of a row in openemr_postcalendar_categories that gives a default visit type (appointment type) for that patient.  It is possible, but rare if a patient of my wife's is seen with two different visit types and these can be easily dealt with in the interface.  Most of her patients are seen for the same reason every time, so this just streamlines the appointment creation process.

Reply:
@sloboy: Correct if wrong -
1. Patient is diagnosed with a condition during a visit - guessing first visit.
2. Patient has 1 or more follow ups to confirm that condition is resolved.
3. No more follow ups are needed, patient records are transferred and patient is marked inactive.

If so, rather than driving the process from patient_data, drive it by diagnosis codes and clinical rules.  Your workflow would be -
1. Create issue with specific diagnosis code(s) - preferably use standard although you can add your own.
2. Create a custom rule with a target to close the issue.  That provides you a mechanism to generate automatic appointment reminders.  You can even have sub cases where diag codes can be tied to frequency of appointments.  Check some of the standard rules (e.g. diabetes).
3. Setup a translation that replaces standard 'deceased' by 'transfer'.  This will provide a standard place for you to confirm transfer date and reason.
4. Create another rule that monitors patients with resolved issues with a target to have transfer date be non-null.  This will provide you standard mechanism to monitor final stage of your patient records.

Reply:
Thank you for the response!  I am going to be revisiting my choices of data fields.  Sorry for the long reply, but I (like I guess every user of Open-EMR thinks) have different characteristics than most users and use cases.  

For example, we won't ever be doing anything at all with Rx or CDC reporting.  Also, her active case load for the present is about 35 weekly appointments (with an end goal of around 150).  She currently has 100 or so patients listed in her spreadsheet that covers the past three years, but that system is not giving her the ability to manage the clinicians.

I am moving in stages on this project.  The first step is getting her data into the system and getting the first step in the workflow started, slowly adding the next steps as she and her clinicians become more used to the changes.  She will  be keeping her current hodgepodge of google drive, sheets, external accounting system while we are doing this project.  The first step is to get appointments scheduled along with attendance and documentation tracking.  From there we move deeper into the system towards integrating billing and accounting.  Does this sound like a good strategy for slowly 'moving in' to Open-EMR?

She currently categorizes the kids by payment source (CMS, private pay, grant) because each of these has characteristics that guide documentation and renewal notices.  The specific diagnosis codes do not affect this process, and all are reimbursed the same amount for that payment source rather than by diagnosis or treatment code.  

Ultimately, these should probably be listed as insurances, but her business is small, and the insurances that are available locally are not accepting new providers for speech therapy.  We don't need the EDI reporting aspects that using the insurance modules would give us.

I need to think about how to move the payment type to the diagnosis codes.  Actually I need to spend some time learning the whole CDR and diagnosis systems.  I need to look at the whole CDR section with diagnosis codes.  This is my first foray into medical information systems, most of my previous experience has been in accounting and management systems.

Thank you for the tip on deceased vs transfer as a translation.  This would be a simpler solution than adding a field.  I am going to look to the existing reminder rules system for the various signpost dates such as reeval, specific reporting dates, etc.  Since I know databases so well, that hammer on my belt is the go-to for solutions sometimes, when I really should keep studying!

Thanks,
Derek

Reply:
Few points as you look under the hood -
1. If you know databases, consider adding views as needed.  Since MySQL reports views as tables, they will show up in rules configuration as custom tables.  For whatever reason, the project does not use views but it is exceptionally efficient and flexible way to implement business logic at database level common across all interfaces.
2. If you realize that codes are attributes and you can have multiple codes per issue and/or encounter, you can see the possibilities in multiple paths allowed by rules.